processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -242,23 +170,111 @@ where
     }
 }
 
-/// Constructs the inclusions to the kernel procedure table.
-fn chiplets_kernel_table_include<E>(main_trace: &MainTrace, alphas: &[E], row: usize) -> E
+/// Builds the inclusions to the kernel procedure table at `row`.
+fn build_kernel_procedure_table_inclusions<E>(main_trace: &MainTrace, alphas: &[E], row: usize) -> E
 where
     E: FieldElement<BaseField = Felt>,
 {
-    if main_trace.is_kernel_row(row) && main_trace.is_addr_change(row) {
-        alphas[0]
-            + alphas[1].mul_base(main_trace.addr(row))
-            + alphas[2].mul_base(main_trace.chiplet_kernel_root_0(row))
-            + alphas[3].mul_base(main_trace.chiplet_kernel_root_1(row))
-            + alphas[4].mul_base(main_trace.chiplet_kernel_root_2(row))
-            + alphas[5].mul_base(main_trace.chiplet_kernel_root_3(row))
+    let selector0 = main_trace.chiplet_selector_0(row);
+    let selector1 = main_trace.chiplet_selector_1(row);
+    let selector2 = main_trace.chiplet_selector_2(row);
+
+    if selector0 == ONE && selector1 == ONE && selector2 == ONE {
```

Comment:
Could we extract `selector0 == ONE && selector1 == ONE && selector2 == ONE` into a function, of the form `is_xyz_chiplet(s0, s1, s2)`? This is difficult to read when you don't know which chiplet `111` references

Reply:
Changed here and in other relevant places